File: SocialAppFlutter/social_app/lib/Models/event_review.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:social_app/Models/user.dart';

import 'event_and_rating_average.dart';

part 'json_generated_code/event_review.g.dart';

@JsonSerializable(explicitToJson: true)
class EventReview {
  final User fromReview;
  final EventAndRatingAverage reviewedEvent;

  EventReview(this.fromReview, this.reviewedEvent);

  factory EventReview.fromJson(Map<String, dynamic> json) =>
      _$EventReviewFromJson(json);

  Map<String, dynamic> toJson() => _$EventReviewToJson(this);
}

File: APIs/EventsService/EventsService/DTOs/EventReview.cs
using Newtonsoft.Json;

namespace EventsService.DTOs
{
    public class EventReview
    {
        [JsonProperty("fromReview")]
        public UserObject FromReview { get; set; } = null!;
        [JsonProperty("reviewedEvent")]
        public EventObjectAndRatingAverage ReviewedEvent { get; set; } = null!;
    }
}
